lib/compiler/src/beam_types.erl

```diff
@@ -1049,6 +1138,13 @@ shrink_union(#t_union{atom=none,list=none,number=none,
 shrink_union(#t_union{atom=none,list=none,number=none,
                       tuple_set=none,other=Other}) ->
     Other;
+shrink_union(#t_union{atom=#t_atom{elements=any},
+                      list=#t_list{type=any,terminator=any},
+                      number=#t_number{elements=any},
+                      tuple_set=#t_tuple{size=0,exact=false,
+                                         elements=Elements},
+                      other=other}) when map_size(Elements) =:= 0 ->
```

Comment:
`map_size(Elements) =:= 0` is redundant since a tuple can't have any element information beyond it's size (see `verified_normal_type/1`.

Reply:
> There's an is_integer/1 test hiding in the first example. :-)

Yes, I need new glasses :(